src/fileio.cpp

```diff
@@ -1166,13 +1183,22 @@ void DeterminePaths(const char *exe)
 {
 	DetermineBasePaths(exe);
 
-#if defined(WITH_XDG_BASEDIR) && defined(WITH_PERSONAL_DIR)
+#if defined(WITH_PERSONAL_DIR) && !defined(_WIN32) && !defined(__APPLE__)
 	char config_home[MAX_PATH];
 
-	const char *xdg_config_home = xdgConfigHome(nullptr);
-	seprintf(config_home, lastof(config_home), "%s" PATHSEP "%s", xdg_config_home,
-			PERSONAL_DIR[0] == '.' ? &PERSONAL_DIR[1] : PERSONAL_DIR);
-	free(xdg_config_home);
+	const char *xdg_config_home = getenv("XDG_CONFIG_HOME");
+	if (xdg_config_home != nullptr) {
+		xdg_config_home = stredup(xdg_config_home);
```

Comment:
why the dup when it is immediately freed?

Reply:
see the comment around getenv("HOME") - it was close, but was separated. I'll duplicate here